src/reducers/fileTree.spec.tsx

```diff
@@ -1012,4 +1017,183 @@ describe(__filename, () => {
+        entries.map((params) =>
+          createInternalVersionEntry({
+            ...fakeVersionEntry,
+            mime_category: 'text',
+            status: 'M',
+            ...params,
+            filename: params.path,
+          }),
+        ),
+      );
+
+      return { pathList, version };
+    };
+
+    it('gets the next path which has a diff', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+        { path: file3, status: 'M' },
+        { path: file4, status: 'M' },
+      ]);
+
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file1,
+          pathList,
+          position: RelativePathPosition.next,
+          version,
+        }),
+      ).toEqual(file3);
+    });
+
+    it('gets the previous path which has a diff', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+        { path: file3, status: 'M' },
+        { path: file4, status: 'M' },
+      ]);
+
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file3,
+          pathList,
+          position: RelativePathPosition.previous,
+          version,
+        }),
+      ).toEqual(file1);
+    });
+
+    it('wraps around to the first path on next', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+        { path: file3, status: 'M' },
+        { path: file4, status: 'M' },
+      ]);
+
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file4,
+          pathList,
+          position: RelativePathPosition.next,
+          version,
+        }),
+      ).toEqual(file1);
+    });
+
+    it('wraps around to the last path on previous', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+        { path: file3, status: 'M' },
+        { path: file4, status: 'M' },
+      ]);
+
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file1,
+          pathList,
+          position: RelativePathPosition.previous,
+          version,
+        }),
+      ).toEqual(file4);
+    });
+
+    it('returns the current path if there are no other paths with changes', () => {
+      const { pathList, version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
+        { path: file2, status: '' },
+      ]);
+
+      expect(
+        findRelativePathWithDiff({
+          currentPath: file1,
+          pathList,
+          position: RelativePathPosition.next,
+          version,
+        }),
+      ).toEqual(file1);
+    });
+
+    it('throws an error if no entry is found for the next path', () => {
+      const { version } = getFakeVersionAndPathList([
+        { path: file1, status: 'M' },
```

Comment:
Wouldn't this be a problem for a version with only one file? An add-on only needs `manifest.json` so it's technically possible.

Reply:
No, but this test's name and setup were a bit misleading, so I fixed that. The error is only thrown when the `pathList` contains a file for which there is not an `entry`. Having just one file is no problem. I updated the test to make it more clear that what we are testing is a `pathList` without a matching `entry`.